src/components/AppProvider/README.md

```diff
@@ -430,13 +430,48 @@ function AppProviderWithAllThemeKeysExample() {
 
 Translations are provided in the locales folder. When using Polaris, you are able to import translations from all languages supported by the core Shopify product and consume them through the `i18n` prop.
 
+At its simplest you can pass the json content from a locale file into `AppProvider`. This approach is suitable when a project has only one specific locale.
+
 ```jsx
+import AppProvider from '@shopify/polaris';
 // en.json is English. Replace with fr.json for French, etc
 import translations from '@shopify/polaris/locales/en.json';
 
-ReactDOM.render(
-  <AppProvider i18n={translations}>{/* App content */}</AppProvider>,
-);
+function App() {
+  return <AppProvider i18n={translations}>{/* App content */}</AppProvider>;
+}
+```
+
+Alternatively if your project supports multiple locales you can load them dynamically using [@shopify/react-i18n](https://github.com/Shopify/quilt/tree/master/packages/react-i18n#translation). This will ensure you only load the translations you need instead of potentially loading every set of translations even if you are not using them.
+
+```jsx
+import AppProvider from '@shopify/polaris';
+// en.json is English. Replace with fr.json for French, etc
+import translations from '@shopify/polaris/locales/en.json';
+import {useI18n} from '@shopify/react-i18n';
+
+function App() {
+  const [i18n] = useI18n({
+    id: 'Polaris',
+    fallback: translations,
+    translations(locale) {
+      return import(
+        /* webpackChunkName: "Polaris-i18n-[request]", webpackMode: "lazy-once" */ `@shopify/polaris/locales/${locale}.json`
```

Comment:
I'm not too familiar with `react-i18n`, does this require specific webpack configuration?

Reply:
Nothing beyond what sewing-kit already provides

Reply:
We may want to add a section here about webpack being required

Reply:
That seems pretty strongly implied. This section is targeted at shopify developers, it's almost guaranteed that they'll be working on a project that uses sewing-kit, and thus webpack and react-i18n will be already preconfigured anyway.